recipe/build.sh

```diff
@@ -113,8 +113,7 @@ python ./configure \
   --with-superlu_dist-include=$PREFIX/include/superlu-dist \
   --with-superlu_dist-lib=-lsuperlu_dist \
```

Comment:
I had the same thought, but while [suitesparse searches include/suitesparse](https://gitlab.com/petsc/petsc/-/blob/v3.21.1/config/BuildSystem/config/packages/SuiteSparse.py#L21), superlu_dist [does not search include/superlu-dist](https://gitlab.com/petsc/petsc/-/blob/v3.21.1/config/BuildSystem/config/packages/SuperLU_DIST.py#L13). Also unlike suitesparse, the prefix is not the default, but [chosen to avoid conflicts](https://github.com/conda-forge/superlu_dist-feedstock/pull/40). Though [debian does the same thing](https://packages.debian.org/bullseye/amd64/libsuperlu-dist-dev/filelist), so it may be worth adding to the default search.

Reply:
I believe `--superlu_dist-dir=$PREFIX` would work after https://gitlab.com/petsc/petsc/-/merge_requests/7551